data/styles/clear/include/Icons.mapcss

```diff
@@ -2535,11 +2541,16 @@ node|z17-[amenity=parking][fee?],
 area|z17-[amenity=parking][fee?],
 {icon-image: parking_pay-m.svg}
 
-node|z17-[amenity=parking][parking=street_side],
-area|z17-[amenity=parking][parking=street_side],
-node|z17-[amenity=parking][parking=lane],
-area|z17-[amenity=parking][parking=lane],
+node|z18-[amenity=parking][parking=street_side],
+area|z18-[amenity=parking][parking=street_side],
+node|z18-[amenity=parking][parking=lane],
+area|z18-[amenity=parking][parking=lane],
 {icon-image: parking-s.svg;}
+node|z18-[amenity=parking][parking=street_side][fee?],
+area|z18-[amenity=parking][parking=street_side][fee?],
+node|z18-[amenity=parking][parking=lane][fee?],
+area|z18-[amenity=parking][parking=lane][fee?],
```

Comment:
all these will not work, unfortunately...
for it to work we'll need new special types added, e.g. in mapcss-mapping.csv:
```
amenity|parking|lane;[amenity=parking][parking=lane][!fee];
amenity|parking|fee|lane;[amenity=parking][fee?][parking=lane];...
```
then a feature with `[amenity=parking][parking=lane][fee=yes]` tags will be matched by 
`amenity|parking|fee|lane` and `amenity|parking|fee` and `amenity|parking`; and the latter two will be eliminated because they're complete prefixes of the first :)
(note that `amenity|parking|lane|fee` will not work)

Reply:
ah i see 👍 fixed :)

Reply:
@pastk That’s interesting. Why would amenity|parking|fee|lane work but amenity|parking|lane|fee wouldn’t? Is there someplace where I can read more about how this works?

Reply:
@dvdmrtnz there is a comment block at the top of mapcss-mapping.csv.

Basically the generator matches all possible lines to feature's given tags and then excludes short matched types which are full prefixes of longer matched types.
E.g. for `[amenity=parking][parking=lane][fee=yes]` following lines/types will match:
`amenity|parking|lane|fee`
`amenity|parking|lane` - will be excluded as a prefix of the above
`amenity|parking|fee` - not a full prefix of `amenity|parking|lane|fee`, so it'll stay
`amenity|parking` - will be excluded as a prefix of the above
So still, we'll end up with `amenity|parking|lane|fee` and `amenity|parking|fee` in the same feature, which we wanted to avoid..
And at the moment its not possible to do e.g. `amenity|parking|fee;[amenity=parking][fee=yes][parking!=lane][parking!=street_side]` (need to add a `!=` support to make it work).

Reply:
@pastk Makes sense! Thanks!